Weather App/view/MainScreenViewController.swift

```diff
@@ -0,0 +1,157 @@
+    var forecastData : [DayForecastCompact] = []
+    var todayForecast : [HourForecastCompact] = []
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        HourlyWeatherCollectionView.delegate = self
+        HourlyWeatherCollectionView.dataSource = self
+        ForecastTableView.delegate = self
+        ForecastTableView.dataSource = self
+        let backGroundImage = UIImageView(frame: UIScreen.main.bounds)
+        backGroundImage.contentMode = .scaleToFill
+        self.view.insertSubview(backGroundImage, at: 0)
+        
+        let blurEffect = UIBlurEffect(style: .systemUltraThinMaterialDark)
+        let blurview = UIVisualEffectView(effect: blurEffect)
+        blurview.frame = UIScreen.main.bounds
+        self.view.insertSubview(blurview, at: 1)
+        
+        openSearchBtn.layer.cornerRadius = 10
+        currentLocationBtn.layer.cornerRadius = 10
+        
+        // data binding
+        currentWeatherViewModel.currentWeatherDescription.bind { [weak self] currentWeatherView in
+            self?.LocationLabel.text = currentWeatherView.location
+            self?.CurrentTempLabel.text = "\(currentWeatherView.temp)°"
+            self?.CurrentWeatherCondn.text = currentWeatherView.condition
+            self?.CurrentLocationLat.text = "H:\(currentWeatherView.lat)°"
+            self?.CurrentLocationLong.text = "L:\(currentWeatherView.long)°"
+        }
+        currentWeatherViewModel.currentAirQuality.bind { [weak self] airQualityView in
+            self?.COValueLabel.text = airQualityView.co
+            self?.NO2ValueLabel.text = airQualityView.no2
+            self?.PM25ValueLabel.text = airQualityView.pm25
+            self?.PM10ValueLabel.text = airQualityView.pm10
+        }
+        currentWeatherViewModel.currentWindDescription.bind { [weak self] windViewModel in
+            self?.SpeedValueLabel.text = windViewModel.speed
+            self?.DegreeValueLabel.text = windViewModel.degree
+            self?.DirectionValueLabel.text = windViewModel.direction
+        }
+        currentWeatherViewModel.forecastedData.bind { [weak self] tempForecastDataList in
+            self?.forecastData = tempForecastDataList
+            self?.ForecastTableView.reloadData()
+        }
+        currentWeatherViewModel.hourforecastToday.bind { [weak self] hourForecastData in
+            self?.todayForecast = hourForecastData
+            self?.HourlyWeatherCollectionView.reloadData()
+        }
+        currentWeatherViewModel.backGroundImage.bind { bkgImage in
+            backGroundImage.image = bkgImage
+        }
+        currentWeatherViewModel.astroToday.bind { [weak self] astrotodayData in
+            self?.sunriseTimeLabel.text = astrotodayData.sunrise
+            self?.sunsetTimeLabel.text = astrotodayData.sunset
+            self?.moonRiseTimeLabel.text = astrotodayData.moonrise
+            self?.moonsetTimeLabel.text = astrotodayData.moonset
+        }
```

Comment:
1. create a separate function name bindViewModel 
2. add this code in bindViewModel. 
3. call bindViewModel form viewDidLoad

Reply:
done